BridgeAgent/BridgeAgentImp.cpp

```diff
@@ -20632,14 +20632,67 @@ void CBridgeAgentImp::ConvertSegmentPathCoordinateToSpanPoint(const CSegmentKey&
    ConvertSegmentCoordinateToSpanPoint(segmentKey,Xs,pSpanKey,pXspan);
 }
 
-void CBridgeAgentImp::GetPointsOfInterestInRange(Float64 xLeft,const pgsPointOfInterest& poi,Float64 xRight,PoiList* pPoiList) const
+void CBridgeAgentImp::GetPointsOfInterestInRange(Float64 xLeft, const pgsPointOfInterest& poi,
```

Comment:
I think this still works with a PoiList. It looks like the only POI that are created on the stack are poiRight and poiLeft. These POI are never stored in the list. All other POIs come from PoiMgr which, before it was changed, return references to POIs.

Reply:
Are you suggesting that the agent should return a PoiList, while the PoiMgr stores POI in a vector? The PoiMgr has to store POI since the POI cannot be referenced for a girder. It can only be referenced for a segment.

I tested Duckabush yesterday and I need to change the implementation in the bridge agent. This is still a work in progress.

Reply:
Yes. PoiMgr has ownership of the POIs so it has to store them in its own vectors. The PoiMgr should be able to fill up a PoiList with references to the POIs that it owns. That's what implementation does.

In the bridge agent, two POIs are created on the stack, but they are not every put into the vector (or PoiList) so it doesn't matter if they go out of scope.